pkg/util/kubernetes/apiserver/metadata.go

```diff
@@ -256,12 +297,7 @@ func (m *MetadataController) deleteMappedEndpoints(namespace, svc string) error
 		metaBundle.Services.Delete(namespace, svc)
 		metaBundle.m.Unlock()
 
-		cacheKey := agentcache.BuildAgentKey(metadataMapperCachePrefix, node.Name)
-		if metaBundle.Empty() {
-			agentcache.Cache.Delete(cacheKey)
-			continue
-		}
-		agentcache.Cache.Set(cacheKey, metaBundle, agentcache.NoExpiration)
```

Comment:
We now only delete the meta bundle in the `deleteNode` handler so that the `metamap` command will still report `detected` for the node.